### DISCUSSION TITLE
3.3.3 broken?

### DISCUSSION THREAD

**Discussion by pauliopolis** [cobbler/cobbler#3538] [Q&A]
I have installed 3.3.3 from EPEL on RHEL 9.3.  Everything looks fine but I don’t seem to be able to get a working PXE installation happening.
Imported a RHEL and a Centos OS but I see the automatic installation metadata field is not being populated.  Client loads kernel and ramdisk as expected then no further progress.  Cobbler log mentions a bunch of missing files culminating with an error that the ‘tree’ cannot be found.

**Comment by SchoolGuy** [comment]
Could it be that you are running into https://github.com/cobbler/cobbler/issues/3417?

**Comment by pauliopolis** [comment]
Not quite - adding more detail

```
cobbler import --path=/mnt --name=rhel-9.3
cobbler distro report --name=rhel9-3-x86_64 | grep Automatic
Automatic Installation Template Metadata : {}
```

QuickStart Guide says it should look like this : 

```
Automatic Installation Template Metadata : {'tree': 'http://@@http_server@@/cblr/links/fedora17-x86_64'}
```

I moved on and did a cobbler sync

```
grep -w append /var/lib/tftpboot/pxelinux.cfg/default
append initrd=/images/rhel-9.3-x86_64/initrd.img kssendmac inst.ks=http://1.2.3.4/cblr/svc/op/autoinstall/profile/rhel-9.3-x86_64
```

curl says that's the sample.ks (even though settings.yml has default.ks)

Anyway moving on and following the pattern in #3417
interrupted the boot to edit 
appended 
inst.repo="http://1.2.4.5/cblr/distro_mirror/rhel-9.3"

result : 
Loading kernel ...
Loading initial ramdisk ...
error: ../../grub-core/net/tftp.c:254:File not found.
...done
press any key to continue

So I guess something about my tftp is not right and/or its the missing tree metadata ????

**Reply by SchoolGuy** [reply]
So two things...

1. The snippet you showed from your distro shows that bug #3417 applies to your scenario.
2. A TFTP error with "File not found" means that either your kernel or your initrd could not be loaded.

To continue debugging this I would kindly ask you to present the Grub config file for the profile/system that Cobbler generated.

**Reply by pauliopolis** [reply]
Can you clarify exactly which file please?

**Reply by pauliopolis** [reply]
mount CentOS-7-x86_64-Minimal-2009.iso /mnt
cobbler import --name=centos-7.9 --path=/mnt
cobbler distro list : centos-7.9-x86_64
cobbler profile list : centos-7.9-x86_64
/etc/cobbler/settings.yaml : autoinstall: "default.ks"

cobbler mkloaders
cobbler sync

/var/lib/tftpboot/pxelinux.cfg/default
DEFAULT menu
PROMPT 0
MENU TITLE Cobbler | https://cobbler.github.io
TIMEOUT 6000
ONTIMEOUT local

LABEL local 
    MENU LABEL (local)
    MENU DEFAULT
    LOCALBOOT -1
 
LABEL centos-7.9-x86_64
    MENU LABEL centos-7.9-x86_64/vmlinuz
    append initrd=/images/centos-7.9-x86_64/initrd.img kssendmac inst.ks=http://1.2.3.4/cblr/svc/op/autoinstall/profile/centos-7.9-x86_64
    ipappend 2
MENU end

Still not sure what you mean by ' the grub config file that cobbler generated'
/var/lib/tftpboot/grub/grub.cfg is just a copy of /var/lib/cobbler/grub_config/grub.cfg provided by the cobbler-3.3.3-1.el9 rpm

**Reply by SchoolGuy** [reply]
So since your system chose to boot grub the `pxelinux.cfg` file is not relevant. The grub file I am interested in is the following: `/var/lib/tftpboot/grub/system/aa:bb:cc:dd:ee:ff`

In case you haven't added a System to Cobbler I strongly recommend you doing so. In this case the content of the following file is of interest: `/var/lib/tftpboot/grub/x86_64_menu_items.cfg`
